proxy.go

```diff
@@ -62,19 +63,21 @@ type Proxy struct {
 
 // NewProxy returns a new HTTP proxy.
 func NewProxy() *Proxy {
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
 	return &Proxy{
 		roundTripper: &http.Transport{
 			// TODO(adamtanner): This forces the http.Transport to not upgrade requests
 			// to HTTP/2 in Go 1.6+. Remove this once Martian can support HTTP/2.
-			TLSNextProto: make(map[string]func(string, *tls.Conn) http.RoundTripper),
-			Proxy:        http.ProxyFromEnvironment,
-			Dial: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-			}).Dial,
+			TLSNextProto:          make(map[string]func(string, *tls.Conn) http.RoundTripper),
+			Proxy:                 http.ProxyFromEnvironment,
+			Dial:                  dialer.Dial,
```

Comment:
I'm worried that resetting the dialer down below won't actually change this since I think it will still be the pointer to the previous Dial function. Instead of passing a dialer throw a `dial func(string, string) (net.Conn, error)` on the `Proxy` struct and let them change that. Use `Dial: net.Dial` here for the default.

Reply:
fixed